web3.js/test/connection.test.ts

```diff
@@ -903,6 +925,132 @@ describe('Connection', function () {
     ).to.be.rejectedWith(mockErrorMessage);
   });
 
+  describe('confirm transaction with mock timer', () => {
+    let clock: SinonFakeTimers;
+    beforeEach(() => {
+      clock = useFakeTimers();
+    });
+
+    afterEach(() => {
+      clock.restore();
+    });
+
+    it('confirm transaction - timeout expired', async () => {
+      const newSignature =
+        'w2Zeq8YkpyB463DttvfzARD7k9ZxGEwbsEw4boEK7jDp3pfoxZbTdLFSsEPhzXhpCcjGi2kHtHFobgX49MMhbWt';
+
+      await mockRpcMessage({
+        method: 'signatureSubscribe',
+        params: [newSignature, {commitment: 'finalized'}],
+        result: new Promise(() => {}),
+      });
+      const timeoutPromise = connection.confirmTransaction(newSignature);
+
+      clock.tick(60 * 1000);
+
+      await expect(timeoutPromise).to.be.rejectedWith(
+        `Signature ${newSignature} has expired: timeout exceeded.`,
+      );
```

Comment:
The idea is that, since you're throwing an actual custom `Error` with a backing class and everything, you should be able to assert that an error _of that type_ has been thrown from the code – not just that the message matches.

```suggestion
      await expect(timeoutPromise).to.be.rejectedWith(TransactionExpiredTimeoutError)
```

Reply:
Fixed now!